packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -707,18 +707,25 @@ class LinearGradient extends Gradient {
       return a.scale(1.0 - t);
     // Interpolation is only possible when the lengths of colors and stops are
     // the same or stops is null for one.
+    // TODO(xster): lerp unsimilar LinearGradients in the future by scaling 
+    // lists of LinearGradients. 
     assert(a.colors.length == b.colors.length);
     assert(a.stops == null || b.stops == null || a.stops.length == b.stops.length);
     final List<Color> interpolatedColors = <Color>[];
-    for (int i = 0; i < a.colors.length; i++) {
-      interpolatedColors.add(Color.lerp(a.colors.elementAt(i), b.colors.elementAt(i), t));
-    }
+    for (int i = 0; i < a.colors.length; i++) 
+      interpolatedColors.add(Color.lerp(a.colors[i], b.colors[i], t));
+    List<double> interpolatedStops;
+    if (a.stops != null && b.stops != null) {
+      for (int i = 0; i < a.stops.length; i++)
+        interpolatedStops.add(ui.lerpDouble(a.stops[i], b.stops[i], t));
+    } else 
+      interpolatedStops = a.stops ?? b.stops;
```

Comment:
nit: yeah, sorry, the style guide is vague (or even technically wrong) about this. the prevailing style is to either have braces on both sides of the else or have them or neither side.